ex4B/ex4.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Write a function that convers a string into number; return null if the string
+ * doesn't contain a valid number.
+ *
+ * For example:
+ * toNumber("71") === 71
+ * toNumber("a7") === null
+ */
+function toNumber(str) {
+
+	var strNum = '', strDec = '', number = null, i = 0, seenDot = false, negative = false;
+	const CODE_FOR_0 = "0".charCodeAt();
```

Comment:
If you move the const outside of the function, you will not need to pass it it the other functions (it's within the closure of the page).

Think of it this way: You modified `strToNumber(str) {...}` `strToNumber(str, CODE_FOR_0) {...}`. Does strToNumber really needs CODE_FOR_0 to do its work? No, because you could have calculated it again if you really wanted to. So, it shouldn't be in the signature of this method, hence it's better to rely on a more-"global" const.